### DISCUSSION TITLE
how to improve HA after setting up RKE2 on 3 server nodes?

### DISCUSSION THREAD

**Discussion by harridu** [rancher/rke2#6556] [Q&A]
For setting up a cluster with 3 server nodes the [documentation](https://docs.rke2.io/install/ha) is a bit wishy-washy, but AFAIU it recommends to setup the first node with this config file:
```
tls-san:
  - rancher01.example.com
  - rancher01a.example.com
  - rancher01b.example.com
  - rancher01c.example.com
```
rancher01 is the common DNS entry for all nodes rancher01{a..c} in this example. Second and third node should be setup with
```
server: https://rancher01a.example.com:9345 
token: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx::server:yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy
tls-san:
  - rancher01.example.com
  - rancher01a.example.com
  - rancher01b.example.com
  - rancher01c.example.com
```
Question, is it reasonable to change all 3 server nodes to
```
server: https://rancher01.example.com:9345 
token: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx::server:yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy
tls-san:
  - rancher01.example.com
  - rancher01a.example.com
  - rancher01b.example.com
  - rancher01c.example.com
```
after first time installation?

**Comment by brandond** [comment]
1. You don't need to add the node names to the TLS SAN list, this is done for you. You only need to add external LB names, or other names/addresses not associated with server nodes.
2. It is fine but not necessary to change the server address on servers after they're started. The only reason you'd really need it there is if you needed to rejoin the cluster for some reason, it is mostly ignored once the cluster has been joined.
3. If you're using a DNS alias for the server address, make sure that it only points to cluster members that are actually functional. Using a DNS server with health checking capability is a good idea.
